allure-java-commons/src/main/java/io/qameta/allure/util/AnnotationUtils.java

```diff
@@ -146,6 +149,35 @@ private AnnotationUtils() {
                 .collect(Collectors.toSet());
     }
 
+    private static Collection<? extends Link> extractLinks(final Collection<Annotation> annotations,
+                                                           final List<Class<? extends Annotation>> ignore) {
+        return annotations.stream()
+                .flatMap(AnnotationUtils::extractRepeatable)
+                .filter(annotation ->
+                        Stream.of(annotation.annotationType().getAnnotationsByType(LabelAnnotation.class))
+                                .anyMatch(it -> it.name().equals("link"))
+                                && !ignore.contains(annotation.annotationType())
+                )
+                .flatMap(annotation -> AnnotationUtils.toLink(annotation).stream())
+                .collect(Collectors.toSet());
+    }
+
+    private static Set<Link> toLink(final Annotation annotation) {
+        try {
+            final Method method = annotation.annotationType().getMethod(VALUE_METHOD_NAME);
+            final Object object = method.invoke(annotation);
+            return objectToStringStream(object)
+                    .map(value -> ResultsUtils.createLink("", value, "", "custom"))
+                    .collect(Collectors.toSet());
+        } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+            LOGGER.error(
+                    "Invalid annotation {}: marker annotations should contains value() method",
```

Comment:
The whole point of LabelAnnotation is that you can create custom annotation per value, for example instead of using `@Feature("Something")` you can use `@Something`. That way it become much easier to maintain and refactor custom files (aka labels) in projects. And from my point of view we should keep it the same way for link annotations

Reply:
So how is it possible to create MyCustomLink? And resolve case described above